### PR TITLE
NET-1847:Add STUN settings

### PR DESCRIPTION
**Pull request by yabinma**
## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [x] Netclient & Netmaker are awesome.

### CODE REVIEW

File: functions/daemon.go [hunk 1]
```diff
@@ -578,8 +591,19 @@ func UpdateKeys() error {
...
-
-	pubIP, pubPort, natType = stun.HolePunch(portToStun, proto)
+	server := config.GetServer(config.CurrServer)
+	if server == nil {
+		server = &config.Server{}
+		server.Stun = true
+		stun.SetDefaultStunServers()
+	}
+	if server.Stun {
+		pubIP, pubPort, natType = stun.HolePunch(portToStun, proto)
```
Comment (abhishek9686) [inline functions/daemon.go:601]: add this check inside stun.HolePunch func and just return if stun is false

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: functions/daemon.go (hunk 1 above)
Comment (yabinma) [reply functions/daemon.go:600]: Done in new commit. please review again.

File: functions/mqhandlers.go [hunk 2]
```diff
@@ -243,6 +244,32 @@ func HostPeerUpdate(client mqtt.Client, msg mqtt.Message) {
...
+		saveServerConfig = true
+		reloadStun = true
+	}
+	if peerUpdate.StunServers != server.StunServers {
+		server.StunServers = peerUpdate.StunServers
+		saveServerConfig = true
+		reloadStun = true
+	}
+
+	if reloadStun {
```
Comment (abhishek9686) [inline functions/mqhandlers.go:259]: why reload here?
 we restarting the daemon here anyway on startup there is code already to load stun servers

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: functions/mqhandlers.go (hunk 2 above)
Comment (yabinma) [reply functions/mqhandlers.go:258]: The restart was added later in the test. 
The load is removed in new commit, only restart here.

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**